### PR TITLE
refactor: update typehints in workflows/cloud-client

### PR DESCRIPTION
**Pull request by Sita04**
## Description

b/280881575 : Update type hints in workflows/cloud-client.

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [x] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Review by kweinmeister — COMMENTED** [review]
Code looks good. Can you please add a Noxfile to enforce type hints? This will test your changes, and will ensure they are enforced in the future. Here's an [example file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/2a633ee36536b0f40ca157bc2a8dd591bf814041/container/snippets/noxfile_config.py#L21) which should work as-is.

**Comment by Sita04** [comment]
Done, thanks Karl.

**Review by kweinmeister — APPROVED** [review]
Code looks good. Can you please add a Noxfile to enforce type hints? This will test your changes, and will ensure they are enforced in the future. Here's an [example file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/2a633ee36536b0f40ca157bc2a8dd591bf814041/container/snippets/noxfile_config.py#L21) which should work as-is.

File: workflows/cloud-client/main.py [hunk 1]
```diff
@@ -48,25 +51,21 @@ def execute_workflow(
     # Wait for execution to finish, then print results.
     execution_finished = False
     backoff_delay = 1  # Start wait with delay of 1 second
-    print('Poll every second for result...')
-    while (not execution_finished):
-        execution = execution_client.get_execution(
-            request={"name": response.name})
+    print("Poll every second for result...")
```
Comment (engelke) [inline workflows/cloud-client/main.py:54]: This is wrong. It's Polling, but not every second. could just say Poll for result instead.

**Review by engelke — APPROVED** [review]
LGTM

**Pull request merged**